### ISSUE TITLE
IA 2.6.7 has broken stream that previously worked

### ISSUE THREAD

**Issue by mbebe** [peak3d/inputstream.adaptive#608]
Hi, 
We have error with IA 2.6.7

```
DecryptSampleData: Decrypt failed with error: 2 and key: 892B1FC3A88E4444A87C66F4EF8E4708
```

Mpd file: http://hls.redefine.pl/E4C29CDF/23986/0/dash_iptv_sd/live.mpd

Previous version was working fine.

```
2021-02-24 22:48:22.579 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: CDMMessage: 2 arrived!
2021-02-24 22:48:22.580 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: Sessionkey:892B1FC3A88E4444A87C66F4EF8E4708 status: 0 syscode: 0
2021-02-24 22:48:22.580 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: CDMMessage: 3 arrived!
2021-02-24 22:48:22.603 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: License update successful
2021-02-24 22:48:22.629 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: CDMMessage: 2 arrived!
2021-02-24 22:48:22.629 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: DecryptSampleData: Decrypt failed with error: 2 and key: 892B1FC3A88E4444A87C66F4EF8E4708
2021-02-24 22:48:22.635 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: Initializing stream with KID: 892B1FC3A88E4444A87C66F4EF8E4708
2021-02-24 22:48:22.636 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: DecryptSampleData: Decrypt failed with error: 2 and key: 892B1FC3A88E4444A87C66F4EF8E4708
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: Reusing DRM psshSets for new period!
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetCapabilities()
2021-02-24 22:48:22.641 T:3932     INFO <general>: Creating Demuxer
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStreamIds()
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1001)
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1002)
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1003)
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1004)
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: CDVDDemuxClient::RequestStream(): added/updated stream 1001 with codec_id 27
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: CDVDDemuxClient::RequestStream(): added/updated stream 1002 with codec_id 27
2021-02-24 22:48:22.641 T:3932    DEBUG <general>: CDVDDemuxClient::RequestStream(): added/updated stream 1003 with codec_id 27
2021-02-24 22:48:22.642 T:3932    DEBUG <general>: CDVDDemuxClient::RequestStream(): added/updated stream 1004 with codec_id 86018
2021-02-24 22:48:22.642 T:3932     INFO <general>: Opening stream: 1001 source: 256
2021-02-24 22:48:22.642 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: OpenStream(1001)
2021-02-24 22:48:22.642 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: Selecting stream with conditions: w: 960, h: 640, bw: 77476207
2021-02-24 22:48:22.643 T:3932    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to select stream!
2021-02-24 22:48:22.643 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1001)
2021-02-24 22:48:22.643 T:3932  WARNING <general>: CVideoPlayer::OpenStream - Unsupported stream 1001. Stream disabled.
2021-02-24 22:48:22.643 T:3932     INFO <general>: Opening stream: 1002 source: 256
2021-02-24 22:48:22.643 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: OpenStream(1002)
2021-02-24 22:48:22.643 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: Selecting stream with conditions: w: 960, h: 640, bw: 77476207
2021-02-24 22:48:22.643 T:3932    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to select stream!
2021-02-24 22:48:22.644 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1002)
2021-02-24 22:48:22.644 T:3932  WARNING <general>: CVideoPlayer::OpenStream - Unsupported stream 1002. Stream disabled.
2021-02-24 22:48:22.644 T:3932     INFO <general>: Opening stream: 1003 source: 256
2021-02-24 22:48:22.644 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: OpenStream(1003)
2021-02-24 22:48:22.644 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: Selecting stream with conditions: w: 960, h: 640, bw: 77476207
2021-02-24 22:48:22.644 T:3932    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to select stream!
2021-02-24 22:48:22.644 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1003)
2021-02-24 22:48:22.644 T:3932  WARNING <general>: CVideoPlayer::OpenStream - Unsupported stream 1003. Stream disabled.
2021-02-24 22:48:22.645 T:3932     INFO <general>: Opening stream: 1004 source: 256
2021-02-24 22:48:22.645 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: OpenStream(1004)
2021-02-24 22:48:22.645 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: Selecting stream with conditions: w: 0, h: 0, bw: 8608467
2021-02-24 22:48:22.645 T:3932    ERROR <general>: AddOnLog: inputstream.adaptive: Unable to select stream!
2021-02-24 22:48:22.645 T:3932    DEBUG <general>: AddOnLog: inputstream.adaptive: GetStream(1004)
```

**Comment by glennguy** [comment]
Hi @mbebe could you also provide the `inputstream.adaptive.license_key` property? So far the mpd doesn't seem to be geoblocked which is good news for anyone wanting to get involved with debugging.
FWIW `DecryptSampleData: Decrypt failed with error: 2 and key: {key}` isn't the issue here, you'll notice all widevine streams will log this a couple of times when first opening the stream (and before actually downloading any content to decode)... this behaviour of course should be fixed at some point.

**Comment by mbebe** [comment]
I don't know if there's any sensitive data in inputstream.adaptive.license_key, because there're device id and authenticate data. :-/
You can get them after login, so below inputstream.adaptive.license_key is without them. .
```
    data=quote('{"jsonrpc":"2.0","id":1,"method":"getWidevineLicense","params":{"cpid":%d,"mediaId":"'%cpid+mediaid+'","sourceId":"'+sourceid+'","keyId":"'+keyid+'","object":"b{SSM}","deviceId":{"type":"other","value":"'+devcid+'"},"ua":"cpgo_www_html5/2","authData":{"sessionToken":"'+authdata+'"}}}')

    LICENSE_URL = 'https://gm2.redefine.pl/rpc/drm/'
    UAcp=  'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.9 Safari/537.36'
    
    play_item.setProperty('inputstream.adaptive.license_key',
                            LICENSE_URL + '|Content-Type=application%2Fjson&Referer=https://go.cyfrowypolsat.pl/&User-Agent=' + quote(UAcp) +
                            '|'+data+'|JBlicense') 
```
There's a limit number of devices connected, so i don't know how could You check this. 
**edit:**
there's license from JSON license response, maybe it'll help :
```
        "object": {
            "status": "OK",
            "license": "CAUSxwUKwQIIAxIQFwW5F8wSBIaLBjM6L3cqjBiCtIKSBSKOAjCCAQoCggEBAJntWzsyfateJO\/DtiqVtZhSCtW8yzdQPgZFuBTYdrjfQFEEQa2M462xG7iMTnJaXkqeB5UpHVhYQCOn4a8OOKkSeTkwCGELbxWMh4x+Ib\/7\/up34QGeHleB6KRfRiY9FOYOgFioYHrc4E+shFexN6jWfM3rM3BdmDoh+07svUoQykdJDKR+ql1DghjduvHK3jOS8T1v+2RC\/THhv0CwxgTRxLpMlSCkv5fuvWCSmvzu9Vu69WTi0Ods18Vcc6CCuZYSC4NZ7c4kcHCCaA1vZ8bYLErF8xNEkKdO7DevSy8BDFnoKEPiWC8La59dsPxebt9k+9MItHEbzxJQAZyfWgkCAwEAAToUbGljZW5zZS53aWRldmluZS5jb20SgAOuNHMUtag1KX8nE4j7e7jLUnfSSYI83dHaMLkzOVEes8y96gS5RLknwSE0bv296snUE5F+bsF2oQQ4RgpQO8GVK5uk5M4PxL\/CCpgIqq9L\/NGcHc\/N9XTMrCjRtBBBbPneiAQwHL2zNMr80NQJeEI6ZC5UYT3wr8+WykqSSdhV5Cs6cD7xdn9qm9Nta\/gr52u\/DLpP3lnSq8x2\/rZCR7hcQx+8pSJmthn8NpeVQ\/ypy727+voOGlXnVaPHvOZV+WRvWCq5z3CqCLl5+Gf2Ogsrf9s2LFvE7NVV2FvKqcWTw4PIV9Sdqrd+QLeFHd\/SSZiAjjWyWOddeOrAyhb3BHMEwg2T7eTo\/xxvF+YkPj89qPwXCYcOxF+6gjomPwzvofcJOxkJkoMmMzcFBDopvab5tDQsyN9UPLGhGC98X\/8z8QSQ+spbJTYLdgFenFoGq47gLwDS6NWYYQSqzE3Udf2W7pzk4ybyG4PHBYV3s4cyzdq8amvtE\/sNSdOKReuHpfQ=",
            "supported_tracks": [],
            "internal_status": 139,
            "message_type": "SERVICE_CERTIFICATE",
            "client_info": [],
            "content_owner": "cyfrowypolsat",
            "content_provider": "cyfrowypolsat",
            "service_version_info": {
                "license_sdk_version": "16.4.2 Built on Feb 16 2021 15:15:05 (1613517266)",
                "license_service_version": "widevine_license_wls_20210210_101970-RC02"
            }
        }
```

**Comment by glennguy** [comment]
@mbebe this should be fixed in #607 , you might be able to test a zip from here: https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-607/1/artifacts

Also could you please rename this issue to suit, maybe "IA 2.6.7 has broken stream that previously worked" or similar?

**Comment by mbebe** [comment]
It works perfectly with IA from that zip.

**Comment by Mariusz89B** [comment]
I can confirm this, working flawlessly. Thank you.

**Comment by matthuisman** [comment]
thats interesting. I thought that fix was going to be HLS only.

@glennguy 
what made you suspect my PR would be the fix for this?
Looks like maybe a comment / log has been deleted from above?

Is the cause because this MPD is using "/url/" paths for its "media" and "initialization"?

**Comment by glennguy** [comment]
That's right. I tested a locally hosted version of the mpd (stripped contentprotection tags) and found the leading slash to be the cause of the issue.

**Comment by matthuisman** [comment]
@phunkyfish this issue was also fixed with https://github.com/xbmc/inputstream.adaptive/pull/607 so can be closed

**Comment by phunkyfish** [comment]
Closed by v2.6.9/10
